install.sh

```
@@ -1,19 +1,31 @@
 
 # Find all dot files then if the original file exists, create a backup
 # Once backed up to {file}.dtbak symlink the new dotfile in place
-for file in $(find . -maxdepth 1 -name ".*" -type f  -printf "%f\n" ); do
-    if [ -e ~/$file ]; then
-        mv -f ~/$file{,.dtbak}
+# Use absolute path to prevent path traversal vulnerabilities
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+
+while IFS= read -r -d '' file; do
+    filename=$(basename "$file")
+    if [ -e ~/$filename ]; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)